src/collector.js

```diff
@@ -0,0 +1,53 @@
+import { default as Message } from 'message';
+
+// "I have a display case ready and waiting for our newest acquistions!"
+// --Taneleer Tivan
+
+
+export default class Collector {
+
+  constructor() {
+    this.errors = [];
+    this.notices = [];
+    this.warnings = [];
+  }
+
+  getMessageList(type) {
+    switch (type) {
+      case 'error':
+        return this.errors;
+      case 'notice':
+        return this.notices;
+      case 'warning':
+        return this.warnings;
+      default:
+        throw new Error(`Invalid message type "${type}"`);
+    }
+  }
+
+  messageCount() {
```

Comment:
Can we use fancy getters here instead? This is essentially the `length` property of a Collection (rather than the `length` property of a specific subset of messages eg `collection.notices.length`). It would be nice if you could do:

```js
var collection = new Collector();
assert.equal(collection.length, 0);
```

Seems more natural; one less API to memorise.